libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java

```
@@ -410,7 +405,8 @@ protected void onPostExecute(TokenEndpointResponse tr) {
                 doLoadPage(true);
             }
             else {
-                doLoadPage(false);
+                onAuthFlowError("jwt_oauth_error", "fail to swap jwt");
+                callback.finish();
```

it will make a toast of the error, instead of fail silently..